src/SqlAsyncCollector.cs

```diff
@@ -232,6 +232,13 @@ private async Task UpsertRowsAsync(IList<T> rows, SqlAttribute attribute, IConfi
                 IEnumerable<string> bracketedColumnNamesFromItem = GetColumnNamesFromItem(rows.First())
                     .Where(prop => !tableInfo.PrimaryKeys.Any(k => k.IsIdentity && string.Equals(k.Name, prop, StringComparison.Ordinal))) // Skip any identity columns, those should never be updated
                     .Select(prop => prop.AsBracketQuotedString());
+                if (!bracketedColumnNamesFromItem.Any())
+                {
+                    string message = $"No property values found in item to upsert. If using query parameters, ensure that the casing of the parameter names and the property names match.";
```

Comment:
please add a test where we check for the exception message to avoid regressing this scenario in the future